samples/client/petstore/cpp-qt5/PetStore/PetApiTests.cpp

```diff
@@ -38,46 +39,59 @@ void PetApiTests::runTests() {
 void PetApiTests::findPetsByStatusTest() {
     OAIPetApi* api = getApi();
 
-    static QEventLoop loop;
+    QEventLoop loop;
     QTimer timer;
     timer.setInterval(14000);
     timer.setSingleShot(true);
 
-    auto validator = [](QList<OAIPet*>* pets) {
+    auto validator = [this](QList<OAIPet*>* pets) {
         foreach(OAIPet* pet, *pets) {
             QVERIFY(pet->getStatus()->startsWith("available") || pet->getStatus()->startsWith("sold"));
         }
+        emit quit();
+    };
+    auto finalizer = [&]() {
         loop.quit();
     };
-
+    connect(this, &PetApiTests::quit, finalizer);
     connect(api, &OAIPetApi::findPetsByStatusSignal, this, validator);
-    connect(&timer, &QTimer::timeout, &loop, &QEventLoop::quit);
+    connect(&timer, &QTimer::timeout, &loop, finalizer);
 
     QList<QString*>* status = new QList<QString*>();
-    status->append(new QString("available"));
-    status->append(new QString("sold"));
+    auto available = new QString("available");
```

Comment:
We will do it in the next update.
I plan to switch to QVariant for all members and lets see. It will be a breaking change but I will make it with the server first.